src/sdam/topology_description.ts

```diff
@@ -400,7 +400,9 @@ function updateRsFromPrimary(
       // replace serverDescription with a default ServerDescription of type "Unknown"
       serverDescriptions.set(
         serverDescription.address,
-        new ServerDescription(serverDescription.address)
+        new ServerDescription(serverDescription.address, undefined, {
+          error: new MongoStalePrimaryError(serverDescription, maxSetVersion, maxElectionId)
```

Comment:
We have been bitten by the risk posed to memory when preserving errors rather than throwing them, should we employ the trick (invoke `stack` getter) to prevent these from holding the scope above in memory or are we confident these are short-lived?

Reply:
Good catch here.